src/lib.rs

```diff
@@ -58,11 +58,17 @@ pub async fn run_trin(
         setup_temp_dir();
     }
 
-    let storage_config =
-        PortalStorageConfig::new(trin_config.kb.into(), discovery.local_enr().node_id());
+    let storage_config = PortalStorageConfig::new(
+        trin_config.kb.into(),
+        discovery.local_enr().node_id(),
+        trin_config.enable_metrics_with_url.is_some(),
+    );
+
+    let mut oracle_storage_config = storage_config.clone();
```

Comment:
There might be a better way to work around this... but for now I'd just add a comment explaining why we're creating this extra copy.